src/game_player.cpp

```diff
@@ -314,84 +320,61 @@ void Game_Player::Update() {
 }
 
 bool Game_Player::CheckActionEvent() {
-	if (InAirship())
+	if (IsFlying()) {
 		return false;
+	}
 
-	// Use | instead of || to avoid short-circuit evaluation
-	return CheckEventTriggerHere({RPG::EventPage::Trigger_action}, true)
-		| CheckEventTriggerThere({RPG::EventPage::Trigger_action,
-		RPG::EventPage::Trigger_touched, RPG::EventPage::Trigger_collision}, true);
-}
-
-bool Game_Player::CheckTouchEvent() {
-	return CheckEventTriggerHere({RPG::EventPage::Trigger_touched});
-}
+	bool result = CheckEventTriggerHere({RPG::EventPage::Trigger_action}, true, true);
 
-bool Game_Player::CheckCollisionEvent() {
-	if (InAirship())
-		return false;
+	int front_x = Game_Map::XwithDirection(GetX(), GetDirection());
+	int front_y = Game_Map::YwithDirection(GetY(), GetDirection());
 
-	return CheckEventTriggerHere({RPG::EventPage::Trigger_collision});
-}
+	result |= CheckEventTriggerThere({RPG::EventPage::Trigger_touched, RPG::EventPage::Trigger_collision}, front_x, front_y, true, true);
 
-bool Game_Player::CheckEventTriggerHere(TriggerSet triggers, bool triggered_by_decision_key) {
-	bool result = false;
+	// Counter tile loop stops only if you talk to an action event.
+	bool got_action = CheckEventTriggerThere({RPG::EventPage::Trigger_action}, front_x, front_y, true, true);
+	// RPG_RT allows maximum of 3 counter tiles
+	for (int i = 0; !got_action && i < 3; ++i) {
+		if (!Game_Map::IsCounter(front_x, front_y)) {
```

Comment:
could you help my memory and tell me what a counter tile was?

Reply:
ah nvm, this was for these tables in e.g. pubs to talk to the shop keeper on the other side.
Somehow confused this with tiles that damage you when entered.

Great find (y)